### PR TITLE
task 2 init api and get data demo

### PR DESCRIPTION
**Pull request by dongsonbk58**

### CODE REVIEW

File: project_training/Api.swift [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+//
+//  Created by Nguyen Dong Son on 4/2/18.
+//  Copyright © 2018 Nguyen Dong Son. All rights reserved.
+//
+
+import Foundation
+import Alamofire
+import SwiftyJSON
+
+public class Api{
```
Comment (TranXuanThien) [inline project_training/Api.swift]: coding convention is wrong, need has space after "Api"
Suggest change name of class is ApiService.

File: project_training/Api.swift [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    
+    init() {
+        
+    }
+    
+    class func ApiRequest(api: String,page: Int, completionHandler: @escaping (_ object: JSON) -> ()) {
+        let pathAPI = Common.base_api+"\(api)"+Common.api_key + "\(page)"
```
Comment (TranXuanThien) [inline project_training/Api.swift]: - coding convention is wrong, need has space before and after "+"
- Common.base_api: You should create a Struct URLs to define all link api

File: project_training/Api.swift [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        Alamofire.request(pathAPI, method: .get, parameters: nil, encoding: JSONEncoding.default).responseJSON { (response) in
+            
+            switch response.result {
+            case .success(let value):
+                if let status = response.response?.statusCode {
+                    switch(status){
```
Comment (TranXuanThien) [inline project_training/Api.swift]: coding convention is wrong, need has space after (status)

File: project_training/Api.swift [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+import Foundation
+import Alamofire
+import SwiftyJSON
+
+public class Api{
+    
+    init() {
+        
+    }
```
Comment (TranXuanThien) [inline project_training/Api.swift]: Not config timeout?

File: project_training/Api.swift [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    
+    class func ApiRequest(api: String,page: Int, completionHandler: @escaping (_ object: JSON) -> ()) {
```
Comment (TranXuanThien) [inline project_training/Api.swift]: func name is wrong coding convention

File: project_training/Api.swift (hunk 1 above)
Comment (dongsonbk58) [reply project_training/Api.swift]: yes..i edited

File: project_training/Api.swift (hunk 2 above)
Comment (dongsonbk58) [reply project_training/Api.swift]: yes..i edited and checked

File: project_training/Api.swift (hunk 5 above)
Comment (dongsonbk58) [reply project_training/Api.swift]: i edited

File: project_training/Api.swift (hunk 4 above)
Comment (dongsonbk58) [reply project_training/Api.swift]: i deleted it

File: project_training/Api.swift (hunk 3 above)
Comment (dongsonbk58) [reply project_training/Api.swift]: yes..sr..i edited

File: project_training/ApiService.swift [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+//  Created by Nguyen Dong Son on 4/2/18.
+//  Copyright © 2018 Nguyen Dong Son. All rights reserved.
+//
+
+import Foundation
+import Alamofire
+import SwiftyJSON
+
+class ApiService {
+    class var apiService: ApiService {
```
Comment (LeVanTuan) [inline project_training/ApiService.swift]: class var apiService: ApiService {
   struct Static {
       static let instance: ApiService = ApiService()
   }
   return Static.instance
}
What does it mean?. why not as this template: static let shared = ApiService()

Comment (dongsonbk58) [reply project_training/ApiService.swift]: i know but i read codeing convetion of swift in https://github.com/framgia/coding-standards/blob/master/eng/swift/coding_convention.md

File: project_training/project_training/ViewController.swift [hunk 7]
```diff
@@ -9,17 +9,21 @@
 import UIKit
 
 class ViewController: UIViewController {
-
+    let apiService = ApiService.apiService
```
Comment (LeVanTuan) [inline project_training/project_training/ViewController.swift]: if you want to use static var of  ApiService class, don't need to define a new "let apiService".  You can use it directly.

Comment (dongsonbk58) [reply project_training/project_training/ViewController.swift]: yes..i edited

**Pull request merged**